src/lib/output/Results.ts

```diff
@@ -100,6 +101,11 @@ export class RunResults implements Results {
 	}
 
 	public async toFormattedOutput(format: OutputFormat, verboseViolations: boolean): Promise<FormattedOutput> {
+		const cacheKey: string = String(format) + '_' + String(verboseViolations);
+		if (this.formattedResultsCache.has(cacheKey)) {
+			return Promise.resolve(this.formattedResultsCache.get(cacheKey));
+		}
```

Comment:
As we continue to split up the result processing into smaller ResultsProcessor instances, caching will be needed so that we do not need to keep reformatting the same results with the same format. This is why I introduced caching now.